src/torznab.ts

```diff
@@ -1,141 +1,247 @@
+
+		if (trackersWithSearchingCaps.length === 0) {
+			throw new CrossSeedError("no working indexers available");
 		}
 	}
 
-	const responses = await Promise.allSettled(
-		urls.map((url) => fetchCaps(url))
-	);
-	const zipped = zip(urls, responses);
-	const unsupported = zipped.filter(
-		([, response]) =>
-			response.value?.caps?.searching?.[0]?.search?.[0]?.$?.available !==
-			"yes"
-	);
-	unsupported
-		.map(([url]) => `${url} is either down or doesn't support searching`)
-		.forEach(logger.warn);
-
-	if (unsupported.length > 0 && unsupported.length === responses.length) {
-		throw new CrossSeedError("no working indexers available");
+	async fetchCaps(url: string | URL): Promise<Caps> {
+		return fetch(this.assembleUrl(url, { t: "caps" }))
+			.then((r) => r.text())
+			.then(xml2js.parseStringPromise)
+			.then(this.parseCaps);
 	}
-}
 
-export async function fetchCaps(url: string | URL): Promise<any> {
-	return fetch(assembleUrl(url, { t: "caps" }))
-		.then((r) => r.text())
-		.then(xml2js.parseStringPromise);
-}
+	parseCaps(xml): Caps {
+		const capsSection = xml?.caps?.searching?.[0];
+		const isAvailable = (searchTechnique) =>
+			searchTechnique?.[0]?.$?.available === "yes";
+		return {
+			search: isAvailable(capsSection?.search),
+			tvSearch: isAvailable(capsSection?.["tv-search"]),
+			movieSearch: isAvailable(capsSection?.["movie-search"]),
+		};
+	}
+
+	async parseResults(text: string): Promise<SearchResult[]> {
+		const jsified = await xml2js.parseStringPromise(text);
+		const items = jsified?.rss?.channel?.[0]?.item;
+		if (!items || !Array.isArray(items)) {
+			return [];
+		}
+
+		return items.map((item) => ({
+			guid: item.guid[0],
+			title: item.title[0],
+			tracker:
+				item?.prowlarrindexer?.[0]?._ ??
+				item?.jackettindexer?.[0]?._ ??
+				item?.indexer?.[0]?._ ??
+				"Unknown tracker",
+			link: item.link[0],
+			size: Number(item.size[0]),
+		}));
+	}
 
-async function parse(text: string): Promise<SearchResult[]> {
-	const jsified = await xml2js.parseStringPromise(text);
-	const items = jsified?.rss?.channel?.[0]?.item;
-	if (!items || !Array.isArray(items)) {
-		return [];
+	getBestSearchTechnique(name: string, caps: Caps): TorznabParams {
+		const extractNumber = (str: string): number =>
+			parseInt(str.match(/\d+/)[0]);
+		const encode = (str) => encodeURIComponent(str).split("%20").join(" ");
+		const mediaType = getTag(name);
+		if (mediaType === MediaType.EPISODE && caps.tvSearch) {
+			const match = name.match(EP_REGEX);
+			return {
+				t: "tvsearch",
+				q: encode(cleanseSeparators(match.groups.title)),
+				season: extractNumber(match.groups.season),
+				ep: extractNumber(match.groups.episode),
+			};
+		} else if (mediaType === MediaType.SEASON && caps.tvSearch) {
+			const match = name.match(SEASON_REGEX);
+			return {
+				t: "tvsearch",
+				q: encode(cleanseSeparators(match.groups.title)),
+				season: extractNumber(match.groups.season),
+			};
+		} else {
+			return {
+				t: "search",
```

Comment:
Why is movie searched not used?

Reply:
I started out with movie enabled but I had a season pack (of a miniseries) which got miscategorized as a movie. Since we're using the same arguments(just q) I figured I'd just fall back to t=search.